Document MatchOnMetadata (#7181)
MicrosoftDocs/visualstudio-docs: docs/msbuild/msbuild-items.md

</Target>
```

#### MatchOnMetadata attribute

> The `MatchOnMetadata` attribute is applicable only to `Remove` attributes that reference other items (e.g. `Remove="@(Compile);@(Content)"`) and instructs the Remove operation to match items based on the values of specified metadata names, instead of matching based on the item values.

Reviewer: ```suggestion
The `MatchOnMetadata` attribute is applicable only to `Remove` attributes that reference other items (for example, `Remove="@(Compile);@(Content)"`) and instructs the Remove operation to match items based on the values of specified metadata names, instead of matching based on the item values.
```

Reviewer: Microsoft style for docs is to avoid Latin abbreviations like e.g.

    <B Remove='@(A)' MatchOnMetadata='M1;M2'/>
  </ItemGroup>

  <Target Name="PrintEvaluation">
    <Message Text="%(B.Identity) M1='%(B.M1)' M2='%(B.M2)' M3='%(B.M3)'" />
  </Target>
</Project>
```

In the above example, item values `b2`, `c2`, and `d2` are removed from item `B` because:
 - `b2` and `c2` from `B` match against `b1` from `A` on `M1=2` and `M2=x`
 - `d2` from `B` matches against `c1` from `A` on `M1=3` and `M2=y`

The `Message` task outputs the following:
```
  a2 M1='x' M2='c' M3='m'
  e2 M1='3' M2='Y' M3='p'
  f2 M1='4' M2='' M3='r'
  g2 M1='' M2='' M3='s'
```

Example usage of `MatchOnMetadata` from the [msbuild common sdk](https://github.com/dotnet/msbuild/blob/808b2ae2a176679d15f8c3299e551a63cb55b799/src/Tasks/Microsoft.Common.CurrentVersion.targets#L5019):
```xml
      <_TransitiveItemsToCopyToOutputDirectory Remove="@(_ThisProjectItemsToCopyToOutputDirectory)" MatchOnMetadata="TargetPath" MatchOnMetadataOptions="PathLike" />
```
The above line removes items from `_TransitiveItemsToCopyToOutputDirectory` that have the same `TargetPath` metadata values from items in `_ThisProjectItemsToCopyToOutputDirectory`

#### MatchOnMetadataOptions attribute

> Specifies the string matching strategy used by `MatchOnMetadata`. Possible values are `CaseSensitive`, `CaseInsensitive`, or `PathLike`. The default value is `CaseSensitive`.

Author: ```suggestion
Specifies the string matching strategy used by `MatchOnMetadata` for matching the metadata values between items (metadata names are always matched case insensitive). Possible values are `CaseSensitive`, `CaseInsensitive`, or `PathLike`. The default value is `CaseSensitive`.
```

> Specifies the string matching strategy used by `MatchOnMetadata`. Possible values are `CaseSensitive`, `CaseInsensitive`, or `PathLike`. The default value is `CaseInsensitive`.

Author: ```suggestion
Specifies the string matching strategy used by `MatchOnMetadata`. Possible values are `CaseSensitive`, `CaseInsensitive`, or `PathLike`. The default value is `CaseSensitive`.
```
